.github/workflows/gradle.yml

```diff
@@ -0,0 +1,74 @@
+name: continuous-integration
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build-on-linux-jdk16:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 16
+        uses: actions/setup-java@v2
+        with:
+          java-version: '16 '
+          distribution: 'adopt'
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+      - name: Build with Gradle and generate the artifacts
+        run: ./gradlew assemble test
+      - uses: actions/upload-artifact@v2
+        with:
+          name: Package
+          path: build/libs
+
+  build-on-linux-jdk11:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Build with Gradle
+        run: ./gradlew build
```

Comment:
It would be nice to also run tests for each JDK + OS combination.

Reply:
Which JDK versions would you consider relevant?

Reply:
Whichever ones we claim support for - today I believe that's 8, 11, and 16?

Reply:
Whichever ones are actively supported by Oracle tends to be a reasonable set of versions to maintain compatibility for. Today that would be 8 (LTS), 11 (LTS), and 16 - the same ones you have here.

I now recall that the Gradle `build` target also runs tests, so what you have here looks good!